hmrc-mongo-play-27/src/main/scala/uk/gov/hmrc/mongo/play/json/formats/MongoUuidFormats.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.mongo.play.json.formats
+
+import org.bson.BsonBinarySubType
+import org.bson.UuidRepresentation
+import org.bson.internal.UuidHelper
+import play.api.libs.json._
+
+import java.util.Base64
+import java.util.UUID
+
+abstract class MongoUuids(uuidRep: UuidRepresentation) {
+  outer =>
+
+  private val encoder = Base64.getEncoder
+  private val decoder = Base64.getDecoder
+
+  private val UuidOldSubtype = BsonBinarySubType.UUID_LEGACY.getValue
+  private val UuidNewSubtype = BsonBinarySubType.UUID_STANDARD.getValue
+
+  private val ChosenSubType =
+    if (uuidRep == UuidRepresentation.JAVA_LEGACY) UuidOldSubtype else UuidNewSubtype
+
+  private def readBase64(str: String, subType: Byte): UUID =
+    UuidHelper.decodeBinaryToUuid(
+      decoder.decode(str),
+      subType,
+      uuidRep
+    )
+
+  private def writeBase64(uuid: UUID): String =
+    encoder.encodeToString(
+      UuidHelper.encodeUuidToBinary(uuid, uuidRep)
+    )
+
+  final val uuidReads: Reads[UUID] =
+    Reads
+      .at[String](__ \ "$binary" \ "subType")
+      .map(_.toByte)
+      .flatMap {
+        case `UuidOldSubtype` =>
```

Comment:
If we do ``case `ChosenSubType` =>`` then we'd get the same behaviour as the registered codec - i.e. fail for a different format. Wondering if aligning it would be better, rather than explaining the difference?
As mentioned before, what would a migration path be though? I suspect the simplest way would be to write an aggregate pipeline using mongo uuid utils - in which case I don't think we need to be concerned with it here.

Reply:
As discussed on Slack with `UuidRepresentation.JAVA_LEGACY` the `UuidHelper` will read both formats, whereas it will actually throw an exception for old UUID data when configured with `UuidRepresentation.STANDARD`